clang/lib/CIR/Dialect/IR/CIRAttrs.cpp

```diff
@@ -295,6 +295,60 @@ LogicalResult IntAttr::verify(function_ref<InFlightDiagnostic()> emitError,
   return success();
 }
 
+//===----------------------------------------------------------------------===//
+// FPAttr definitions
+//===----------------------------------------------------------------------===//
+
+Attribute cir::FPAttr::parse(AsmParser &parser, Type odsType) {
+  double value;
```

Comment:
Can't this be tablegen'd? If not entirely, at least parts seems doable with the use of `custom` stuff, see https://mlir.llvm.org/docs/DefiningDialects/AttributesAndTypes/#assembly-format-directives

Reply:
Updated.